### ISSUE TITLE
Error: sp_before_byref=remove does not always take effect when sp_arith=add is used

### ISSUE THREAD

**Issue by arneforlie** [uncrustify/uncrustify#1709]
I don't think this has every worked, but users have started complaining, so now it is reported:

It turns out that when sp_arith=add is used together with sp_before_byref=remove, the removal of space before & does not work if there are two levels of namespaces.

Test case:
```
class T;

namespace Foo
{
void bar(T& t);

namespace Bar
{
void bar(T& t);
}
}

void
Foo::bar(T& t)
{
}

void
Foo::Bar::bar(T& t)
{
}
```
Generated result:
```
class T;

namespace Foo
{
void bar(T& t);

namespace Bar
{
void bar(T& t);
}
}

void
Foo::bar(T& t)
{
}

void
Foo::Bar::bar(T & t)
{
}
```
Expected result:
```
class T;

namespace Foo
{
void bar(T& t);

namespace Bar
{
void bar(T& t);
}
}

void
Foo::bar(T& t)
{
}

void
Foo::Bar::bar(T& t)
{
}
```
The bug is that we get "T space & space t" for Foo::Bar::bar(), but not for Foo::bar()

One additional observation;
if I replace T with int, float or std::string, it works as expected. So there is something funny about user defined types that are not namespace qualified.

Debug: https://pastebin.com/aCB4DPGJ

./build/uncrustify -v
Uncrustify-0.66.1-493-34c4833

**Comment by guy-maurel** [comment]
The current git version works pretty fine.
  https://github.com/uncrustify/uncrustify
Please test it and report!

**Comment by arneforlie** [comment]
Indeed it works.
It turns out that this was fixed by commit bf0c381 about 8 weeks ago.
